careconnect2025/backend/core/src/main/java/com/careconnect/service/USPSDigestService.java

```diff
@@ -0,0 +1,86 @@
+package com.careconnect.service;
+
+import com.careconnect.model.*;
+import com.careconnect.repository.EmailCredentialRepo;
+import com.careconnect.repository.USPSDigestCacheRepo;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.*;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class USPSDigestService {
+    private final EmailCredentialRepo credRepo;
+    private final USPSDigestCacheRepo cacheRepo;
+    private final GmailClient gmailClient;
+    private final OutlookClient outlookClient;
+    private final GmailParser gmailParser;
+    private final OutlookParser outlookParser;
+    private final ObjectMapper om = new ObjectMapper();
+
+    public Optional<USPSDigest> latestForUser(String userId) {
+        // 1) cache
+        var cached = cacheRepo.findFirstByUserIdAndExpiresAtAfterOrderByDigestDateDesc(userId, Instant.now());
```

Comment:
Whats the point in adding all these repo calls if there is no creds in the db and no way for the user to add/create their credentials?